docs/src/architecture/backend_architecture.md

```diff
@@ -0,0 +1,220 @@
+    MatchingSupervisor->>Player1: Session ID (Encoded PID)
+```
+
+### Join game
+
+```mermaid
+sequenceDiagram
+    participant Player2
+    participant Server
+    participant MatchingSupervisor
+    participant MatchingSession
+    Player2->>Server: List lobbies
+    Server->>MatchingSupervisor: list_session_ids()
+    MatchingSupervisor->>Player2: [TKSvPBC9iiXaKrqxigozgYLja71X2s9txQPPqDznj, ...]
+    Player2->>Server: Join Lobby TKSvPBC9iiXaKrqxigozgYLja71X2s9txQPPqDznj
+    Server->>MatchingSession: call(matchmaking_session_pid, :add_player, player)
+    MatchingSession->>Server: OK
+    Server->>Player2: Lobby joined
+```
+
+### Start game
+
+```mermaid
+sequenceDiagram
+    Player1->>Server: Start game TKSvPBC9iiXaKrqxigozgYLja71X2s9txQPPqDznj
+    Server->>MatchingSession: call(matchmaking_session_pid, :start_game)
+    MatchingSession->>MatchingSession: Broadcast {:game_started, matchmaking_session_id, players}
+    MatchingSession->>MatchingSession: Terminate
+```
+
+## Gameplay
+
+When a game starts, two things happen:
+
+- A game session is spawned by a `DynamicSupervisor` called `GameBackend`.  This `GameBackend` starts a new child `Runner` process, which holds the entire game's state and the logic to update it according to the players' actions. The PID of this `Runner` is encoded in a human friendly format and called the `game_session_id`.
+- Every player connects to the game through websocket under the `/play/:game_session_id` path. Each player's connection is handled by a separate [cowboy websocket](https://ninenines.eu/docs/en/cowboy/2.6/manual/cowboy_websocket/) process, defined in the `PlayWebSocket` module. On startup, the process saves the runner's PID so it can communicate with it. Inside the game, a player is the same as a websocket connection.
+
+Let's go over the main gameplay flow. Let's say `player_1` wants to move to the right one square. To do this, they send a `JSON` frame over the socket that looks like this:
+
+```json
+{"action": "move_with_joystick", "value": "{angle}"}
+```
+
+The corresponding `PlayWebSocket` process picks it up, decodes it, then sends a message to the `Runner` with the player's action like this:
+
+```elixir
+GenServer.cast(runner_pid, {:move, user_id, action})
+```
+
+The `Runner`'s appropriate handler eventually picks up this message, which in this case looks like this:
+
+```elixir
+  def handle_cast({:move, user_id, %ActionOk{value: value, timestamp: timestamp}}, state) do
+    angle =
+      case Nx.atan2(value.y, value.x) |> Nx.multiply(Nx.divide(180.0, Nx.Constants.pi())) |> Nx.to_number() do
+        pos_degree when pos_degree >= 0 -> pos_degree
+        neg_degree -> neg_degree + 360
+      end
+
+    player_id = state.user_to_player[user_id]
+    game_state = GameBackend.move_player(state.game_state, player_id, angle)
+
+    state =
+      Map.put(state, :game_state, game_state)
+      |> put_in([:player_timestamps, player_id], timestamp)
+
+    {:noreply, state}
+  end
+```
```

Comment:
I think this is outdated. We're not using ActionOk anymore and it's already receiving the movement angle instead of x and y coordinates